Open telemetry (#22)
CovidShield/server: CONTRIBUTING.md

# Contributing Guidelines

> Welcome to CovidShield. Thank you for considering contributing to COVID Shield!

Reviewer: ~CovidShield~ COVID Shield.

However, it seems like `Welcome to CovidShield` is redundant with the next sentence?

We’d love to get your issues (if you find any bugs) and PRs (if you have any fixes)!

- [Code of Conduct](#code-of-conduct)
- [Reporting Security Issues](#reporting-security-issues)
- [Contributing](#contributing)
  - [Contributing Documentation](#contributing-documentation)
  - [Contributing Code](#contributing-code)

> - [Running CovidShield](#running-covidshield)

Reviewer: ```suggestion
- [Running COVID Shield](#running-covidshield)
```

Reviewer 2: ty

Once you're happy with your changes, please fork the repository and push your
code to your fork, then open a PR against this repository.

> ## Running CovidShield

Reviewer: ```suggestion
## Running COVID Shield
```

Note that 302 is a [MCC](https://www.mcc-mnc.com/): 302 represents Canada.

## Testing

If you're at Shopify, `dev up` will configure the database for you. If not

> you will need to point to your database server using the environment variables

Reviewer: Did we want to remove all Shopify-specific tooling from this repo? There was talks of it before.

Cc: @burke

Reviewer 3: Yeah it probably does make sense to st this point.

```shell
$ make test
```

## Metrics and Tracing

> CovidShield uses [OpenTelemetry](https://github.com/open-telemetry/opentelemetry-go) to configure the metrics and tracing for the server, both the key retrieval and key submission.

Reviewer: ```suggestion
COVID Shield uses [OpenTelemetry](https://github.com/open-telemetry/opentelemetry-go) to configure the metrics and tracing for the server, both the key retrieval and key submission.
```

### Metrics

Currently, the following options are supported for enabling Metrics:
* standard output
* prometheus

Metrics can be enabled by setting the `METRIC_PROVIDER` variable to either `stdout`, or `prometheus`.

If you want to use Prometheus, please see the additional configuration requirements below.

> 

Reviewer 3: ```suggestion
Note that logs are emitted to `stderr`, so with `stdout` mode, logs will be on `stderr` and metrics will be on `stdout`.
```

#### Prometheus 

In order to use Prometheus as a metrics solution, you'll need to be running it in your environment. 

You can follow the instructions [here](https://prometheus.io/docs/prometheus/latest/installation/) for running Prometheus. 

> You will need to edit the configuration file, `prometheus.yml` to add an additional target so it actually polls the metrics coming from the CovidShield server:

Reviewer: ```suggestion
You will need to edit the configuration file, `prometheus.yml` to add an additional target so it actually polls the metrics coming from the COVID Shield server:
```
